e2e/_suites/ingest-manager/fleet.go

```diff
@@ -674,37 +1038,107 @@ func getAgentDefaultPolicy() (string, error) {
 			"error":        err,
 			"responseBody": body,
 		}).Error("Could not parse response into JSON")
-		return "", err
+		return nil, err
 	}
 
 	// data streams should contain array of elements
 	policies := jsonParsed.Path("items")
 
 	log.WithFields(log.Fields{
 		"count": len(policies.Children()),
-	}).Debug("Fleet policies retrieved")
+	}).Trace("Fleet policies retrieved")
+
+	// TODO: perform a strong check to capture default policy
+	defaultPolicy := policies.Index(0)
+
+	return defaultPolicy, nil
+}
+
+func getAgentEvents(applicationName string, agentID string, packagePolicyID string, updatedAt string) error {
+	url := fmt.Sprintf(fleetAgentEventsURL, agentID)
+	getReq := createDefaultHTTPRequest(url)
+	getReq.QueryString = "page=1&perPage=20"
+
+	body, err := curl.Get(getReq)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"agentID":         agentID,
+			"application":     applicationName,
+			"body":            body,
+			"error":           err,
+			"packagePolicyID": packagePolicyID,
+			"url":             url,
+		}).Error("Could not get agent events from Fleet")
+		return err
+	}
+
+	jsonResponse, err := gabs.ParseJSON([]byte(body))
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error":        err,
+			"responseBody": body,
+		}).Error("Could not parse response into JSON")
+		return err
+	}
+
+	listItems := jsonResponse.Path("list").Children()
+	for _, item := range listItems {
+		message := item.Path("message").Data().(string)
+		timestamp := item.Path("timestamp").Data().(string)
+
+		log.WithFields(log.Fields{
+			"agentID":         agentID,
+			"application":     applicationName,
+			"event_at":        timestamp,
+			"message":         message,
+			"packagePolicyID": packagePolicyID,
+			"updated_at":      updatedAt,
+		}).Trace("Event found")
+
+		matches := (strings.Contains(message, applicationName) &&
+			strings.Contains(message, "["+agentID+"]: State changed to") &&
+			strings.Contains(message, "Protecting with policy {"+packagePolicyID+"}"))
```

Comment:
This could be simplified into a regex, but with this I wanted to make it simpler and cleaner to understand what we want to check in the message of an event.

In a previous conversation/comment with @EricDavisX we agreed in having another step, or refactor the one that calls this code, to represent the `success` status of the event. We will probably address it in a follow up PR